pkg/vault/contracts/ProtocolFeeController.sol

```diff
@@ -412,6 +450,59 @@ contract ProtocolFeeController is
         // Allow tracking pool fee percentages in all cases (e.g., when the pool is protocol-fee exempt).
         emit InitialPoolAggregateSwapFeePercentage(pool, aggregateSwapFeePercentage, protocolFeeExempt);
         emit InitialPoolAggregateYieldFeePercentage(pool, aggregateYieldFeePercentage, protocolFeeExempt);
+
+        if (poolCreator != address(0)) {
+            emit PoolWithCreatorRegistered(pool, poolCreator, protocolFeeExempt);
+        }
```

Comment:
!!!!
doesn't this brick pool registration when there's a pool creator? This is set by the vault: https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/VaultExtension.sol#L260-L261

Reply:
Btw how does this not make any test fail?

Reply:
I think I don't understand what you mean here; it's just emitting an extra event now.

Reply:
lol I'm too used to the `if / revert` pattern; my brain read `revert` instead of `emit`.

Do we need to emit this conditionally? Why not just emit an event with the pool creator address?